zio-http/src/main/scala/zio/http/Response.scala

```diff
@@ -38,10 +41,7 @@ final case class Response private (
    * detect the changes and encode the response again, however it will turn out
    * to be counter productive.
    */
-  def freeze(implicit trace: Trace): Task[Response] =
-    for {
-      encoded <- encode()
-    } yield self.copy(attribute = self.attribute.withEncodedResponse(encoded, self))
+  def freeze: Response = self.copy(frozen = true)
```

Comment:
`freeze` is not a pure function any more.

Reply:
The fact that operation only sets a boolean flag would have me disagree with that statement.

As for its semantic meaning, that of caching a static value for performance purposes is executed at the "end of the world".   In this case during the execution of HTTP and ZIO DSLs when a request is handled by Netty.

Would you mind expanding on your statement, maybe I am misunderstanding your concerns?

Reply:
I have removed the `AtomicReference` that was most likely the concern.

The new PoC uses a `ConcurrentMap` to hold frozen responses. 

There may or may not be a need to look at a more complete caching approach though I don't see `frozen`  Responses as a very common uses-cases outside of some simple, static content.